lib/tic_tac_toe.rb

```diff
@@ -59,5 +60,95 @@ def board_full?
       !self.include?(:available)
     end
 
+    def best_move
+      if cpu_check_for_wins_or_blocks_horizontally?(:player_one)
+      elsif cpu_check_for_wins_or_blocks_vertically?(:player_one)
+      elsif cpu_check_for_wins_or_blocks_diagonally?(:player_one)
+  
+      elsif cpu_check_for_wins_or_blocks_horizontally?(:player_two)
+      elsif cpu_check_for_wins_or_blocks_vertically?(:player_two)
+      elsif cpu_check_for_wins_or_blocks_diagonally?(:player_two)
+      else
+         cpu_take_open_space()
+      end
+      @best_move
```

Comment:
```suggestion
         move = cpu_check_for_wins_or_blocks_horizontally?(:player_one)
         move ||= cpu_check_for_wins_or_blocks_vertically?(:player_one)
         move ||= cpu_check_for_wins_or_blocks_diagonally?(:player_one)
  
         move ||= cpu_check_for_wins_or_blocks_horizontally?(:player_two)
         move ||= cpu_check_for_wins_or_blocks_vertically?(:player_two)
         move ||= cpu_check_for_wins_or_blocks_diagonally?(:player_two)
         move ||= cpu_take_open_space()

```
Instead of doing a bunch of if and having to use a class variable. You can change your other methods to return the best_move or nil . instead of true and false. 

the ` ||=` (or equal)  operator  will set the variable if it is currently `nil` if it is already set it does nothing.